central/networkgraph/service/service_impl_postgres_test.go

```diff
@@ -0,0 +1,517 @@
+		s.NoError(err)
+	}
+
+	deploymentToEntity1 := externalFlow(deployment, entity, false)
+	deploymentToEntity3 := externalFlow(deployment, entity3, false)
+
+	deployment2ToEntity3 := externalFlow(deployment2, entity3, false)
+
+	// flow from deployment to es1a and es1c but not es1b
+	flows := []*storage.NetworkFlow{
+		deploymentToEntity1, deploymentToEntity3,
+		deployment2ToEntity3,
+	}
+
+	flowStore, err := s.flowDataStore.CreateFlowStore(globalWriteAccessCtx, testCluster)
+	s.NoError(err)
+
+	err = flowStore.UpsertFlows(globalWriteAccessCtx, flows, timestamp.FromGoTime(time.Now()))
+	s.NoError(err)
+
+	for _, tc := range []struct {
+		name     string
+		request  *v1.GetExternalNetworkFlowsRequest
+		expected *v1.GetExternalNetworkFlowsResponse
+		pass     bool
+	}{
+		{
+			name: "Get single entity flows",
+			request: &v1.GetExternalNetworkFlowsRequest{
+				ClusterId: testCluster,
+				EntityId:  entityID.String(),
+				Query:     fmt.Sprintf("Namespace:%s", testNamespace),
+			},
+			expected: &v1.GetExternalNetworkFlowsResponse{
+				Entity: entity.GetInfo(),
+				Flows: []*storage.NetworkFlow{
+					deploymentToEntity1,
+				},
+			},
+			pass: true,
+		},
+		{
+			name: "Entity with no flows",
+			request: &v1.GetExternalNetworkFlowsRequest{
+				ClusterId: testCluster,
+				EntityId:  entityID2.String(),
+				Query:     fmt.Sprintf("Namespace:%s", testNamespace),
+			},
+			expected: &v1.GetExternalNetworkFlowsResponse{
+				Entity: entity2.GetInfo(),
+				Flows:  []*storage.NetworkFlow{},
+			},
+			pass: true,
+		},
+		{
+			name: "Invalid entity ID",
+			request: &v1.GetExternalNetworkFlowsRequest{
+				ClusterId: testCluster,
+				EntityId:  "invalid ID",
+				Query:     fmt.Sprintf("Namespace:%s", testNamespace),
+			},
+			expected: nil,
+			pass:     false,
+		},
+		{
+			name: "Invalid cluster",
+			request: &v1.GetExternalNetworkFlowsRequest{
+				ClusterId: "invalid cluster",
+				EntityId:  entityID.String(),
+				Query:     fmt.Sprintf("Namespace:%s", testNamespace),
+			},
+			expected: nil,
+			pass:     false,
+		},
+		{
+			name: "entity with multiple flows",
+			request: &v1.GetExternalNetworkFlowsRequest{
+				ClusterId: testCluster,
+				EntityId:  entityID3.String(),
+				Query:     fmt.Sprintf("Namespace:%s", testNamespace),
+			},
+			expected: &v1.GetExternalNetworkFlowsResponse{
+				Entity: entity3.GetInfo(),
+				Flows: []*storage.NetworkFlow{
+					deploymentToEntity3,
+					deployment2ToEntity3,
+				},
+			},
+			pass: true,
+		},
+	} {
+		s.Run(tc.name, func() {
+			response, err := s.service.GetExternalNetworkFlows(ctx, tc.request)
+			if tc.pass {
+				s.NoError(err)
+				protoassert.Equal(s.T(), tc.expected.Entity, response.Entity)
+				protoassert.ElementsMatch(s.T(), tc.expected.Flows, response.Flows)
+			} else {
+				s.Error(err)
+			}
+		})
+	}
+}
+
+func (s *networkGraphServiceSuite) TestGetExternalNetworkFlowsMetadata() {
```

Comment:
Some of my comments on `TestGetExternalNetworkFlows` apply to this test as well.

Reply:
Done